src/web/app/pages-instructor/instructor-session-individual-extension-page/instructor-session-individual-extension-page.component.ts

```diff
@@ -0,0 +1,528 @@
+        this.instructorsOfCourse = instructorModels;
+        this.initialSortOfInstructors();
+      }, (resp: ErrorMessageOutput) => {
+          this.hasLoadedAllInstructorsFailed = true;
+          this.statusMessageService.showErrorToast(resp.error.message);
+        },
+      );
+  }
+
+  private mapInstructorsToInstructorModels(instructors: Instructor[]): InstructorExtensionTableColumnModel[] {
+    return instructors.map((instructor) => {
+      const instructorData: InstructorExtensionTableColumnModel = {
+        name: instructor.name,
+        role: instructor.role,
+        email: instructor.email,
+        extensionDeadline: this.feedbackSessionEndingTimestamp,
+        hasExtension: false,
+        isSelected: false,
+      };
+
+      if (instructor.email in this.instructorDeadlines) {
+        instructorData.hasExtension = true;
+        instructorData.extensionDeadline = this.instructorDeadlines[instructor.email];
+      }
+      return instructorData;
+    });
+  }
+
+  private initialSortOfInstructors(): void {
+    this.instructorsOfCourse.sort(this.sortInstructorPanelsBy(SortBy.INSTRUCTOR_PERMISSION_ROLE));
+    this.instructorsOfCourse.sort(this.sortInstructorPanelsBy(SortBy.SESSION_END_DATE));
+  }
+
+  /**
+   * Handles the the date and time selection modal to create/update deadlines.
+   */
+  onExtend(): void {
+    const modalRef: NgbModalRef = this.ngbModal.open(IndividualExtensionDateModalComponent);
+    modalRef.componentInstance.numStudents = this.getNumberOfSelectedStudents();
+    modalRef.componentInstance.numInstructors = this.getNumberOfSelectedInstructors();
+    modalRef.componentInstance.feedbackSessionEndingTimestamp = this.feedbackSessionEndingTimestamp;
+    modalRef.componentInstance.feedbackSessionTimeZone = this.feedbackSessionTimeZone;
+    modalRef.componentInstance.onConfirmCallBack.subscribe((extensionTimestamp: number) => {
+      this.onConfirmExtension(extensionTimestamp);
+      modalRef.close();
+    });
+  }
+
+  /**
+   * Handles the opening the confirmation modal to create/update deadlines.
+   */
+  private onConfirmExtension(extensionTimestamp: number): void {
+    const modalRef: NgbModalRef = this.ngbModal.open(IndividualExtensionConfirmModalComponent);
+    const selectedStudents = this.getSelectedStudents();
+    const selectedInstructors = this.getSelectedInstructors();
+    modalRef.componentInstance.modalType = ExtensionModalType.EXTEND;
+    modalRef.componentInstance.selectedStudents = selectedStudents;
+    modalRef.componentInstance.selectedInstructors = selectedInstructors;
+    modalRef.componentInstance.extensionTimestamp = extensionTimestamp;
+    modalRef.componentInstance.feedbackSessionTimeZone = this.feedbackSessionTimeZone;
+
+    modalRef.componentInstance.onConfirmExtensionCallBack.subscribe((isNotifyDeadlines: boolean) => {
+      this.handleCreateAndDeleteDeadlines(selectedStudents, selectedInstructors, DeadlineHandlerType.CREATE,
```

Comment:
nit: can line break before `DeadlineHandlerType.CREATE` to highlight the different param input from line 308